apps/frontend/src/pages/public-profile/public-profile.tsx

```diff
@@ -0,0 +1,90 @@
+import { Avatar, Button, Loader } from "~/libs/components/components.js";
+import { DataStatus } from "~/libs/enums/data-status.enum.js";
+import {
+	useAppDispatch,
+	useAppSelector,
+	useCallback,
+	useEffect,
+	useParams,
+} from "~/libs/hooks/hooks.js";
+import { actions as usersActions } from "~/modules/users/users.js";
+
+import styles from "./styles.module.css";
+
+const PublicProfile = (): React.JSX.Element => {
+	const SINGLE_FOLLOWER_COUNT = 1;
+
+	const dispatch = useAppDispatch();
+	const { id } = useParams();
+
+	const authenticatedUser = useAppSelector(
+		({ auth }) => auth.authenticatedUser,
+	);
+	const dataStatus = useAppSelector(({ users }) => users.dataStatus);
+	const userProfile = useAppSelector(({ users }) => users.userProfile);
+
+	useEffect(() => {
+		if (id) {
+			void dispatch(usersActions.getUserPublicProfile(Number(id)));
+		}
+	}, [dispatch, id]);
+
+	const handleFollowToggle = useCallback((): Promise<void> => {
+		if (userProfile && authenticatedUser) {
+			if (userProfile.isFollowed) {
+				void dispatch(
+					usersActions.unfollowUser({
+						followerId: authenticatedUser.id,
+						userId: Number(id),
+					}),
+				);
+			} else {
+				void dispatch(
+					usersActions.followUser({
+						followingId: Number(id),
+						userId: authenticatedUser.id,
+					}),
+				);
+			}
+		}
+
+		return Promise.resolve();
```

Comment:
no need to return promise here
```suggestion
	const handleFollowToggle = useCallback((): void => {
		if (userProfile && authenticatedUser) {
			if (userProfile.isFollowed) {
				void dispatch(
					usersActions.unfollowUser({
						followerId: authenticatedUser.id,
						userId: Number(id),
					}),
				);
				return;
			} 
			
				void dispatch(
					usersActions.followUser({
						followingId: Number(id),
						userId: authenticatedUser.id,
					}),
				);
			}
```